runtime/server/generate_resolver.go

```diff
@@ -0,0 +1,236 @@
+package server
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime"
+	"github.com/rilldata/rill/runtime/drivers"
+	"github.com/rilldata/rill/runtime/pkg/activity"
+	"github.com/rilldata/rill/runtime/pkg/observability"
+	"github.com/rilldata/rill/runtime/queries"
+	"github.com/rilldata/rill/runtime/server/auth"
+	"go.opentelemetry.io/otel/attribute"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+func (s *Server) GenerateResolver(ctx context.Context, req *runtimev1.GenerateResolverRequest) (*runtimev1.GenerateResolverResponse, error) {
+	observability.AddRequestAttributes(ctx,
+		attribute.String("args.instance_id", req.InstanceId),
+		attribute.String("args.connector", req.Connector),
+		attribute.String("args.table", req.Table),
+		attribute.String("args.metrics_view", req.MetricsView),
+		// attribute.String("args.prompt", req.Prompt), // Adding this might be a privacy issue
+	)
+	s.addInstanceRequestAttributes(ctx, req.InstanceId)
+
+	// Must have edit permissions on the repo
+	if !auth.GetClaims(ctx).CanInstance(req.InstanceId, auth.EditRepo) {
+		return nil, ErrForbidden
+	}
+
+	connector := req.Connector
+	if connector == "" {
+		// Get instance
+		inst, err := s.runtime.Instance(ctx, req.InstanceId)
+		if err != nil {
+			return nil, err
+		}
+
+		connector = inst.ResolveOLAPConnector()
+	}
+
+	// Connect to connector and check it's an OLAP db
+	handle, release, err := s.runtime.AcquireHandle(ctx, req.InstanceId, connector)
+	if err != nil {
+		return nil, err
+	}
+	defer release()
+	olap, ok := handle.AsOLAP(req.InstanceId)
+	if !ok {
+		return nil, status.Error(codes.InvalidArgument, "connector is not an OLAP connector")
+	}
+
+	dialect := olap.Dialect().String()
+
+	ctrl, err := s.runtime.Controller(ctx, req.InstanceId)
+	if err != nil {
+		return nil, err
+	}
+
+	if req.Table != "" && req.Connector != "" {
+		// Get table info
+		tbl, err := olap.InformationSchema().Lookup(ctx, req.Table)
+		if err != nil {
+			return nil, status.Error(codes.InvalidArgument, "table not found")
+		}
+
+		start := time.Now()
+
+		sql, err := s.generateResolverForTable(ctx, req.InstanceId, req.Prompt, tbl.Name, dialect, tbl.Schema)
```

Comment:
Would suggest returning `resolver, resolverProps, err` here instead of `sql` (otherwise the function should be `generateSQLResolverForTable`)